### PR TITLE
Query assets by location

### PR DESCRIPTION
**Pull request by tilley14**
Added 3 types of geo-spatial querying.
1. Getting assets within a set distance from a center point
2. Getting assets that are in a rectangle formed by two points (the bottom left and top right)
3. Getting assets that fall within a polygon formed by an array of points.

Also, fixed latitudes and longitudes being treated as the incorrect axis for points. Latitudes are now all Y. Longitudes are now all X.

### CODE REVIEW

File: db/api/v1/db/assets.db.js [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-JOIN project ON asset.project_id = project.id
-JOIN sponsor ON project.sponsor_id = sponsor.id
-JOIN asset_type ON asset.asset_type_id = asset_type.id
-WHERE TRUE `;
+    asset
+    JOIN project ON asset.project_id = project.id
+    JOIN sponsor ON project.sponsor_id = sponsor.id
+    JOIN asset_type ON asset.asset_type_id = asset_type.id
+WHERE 
+    TRUE `;
```
Comment (APalumbo97) [inline db/api/v1/db/assets.db.js]: What is the point of `WHERE TRUE` at the end? Could it be removed?

File: db/api/v1/controllers/geometrySearch.controller.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+};
+
+module.exports = {
+    distanceFind,
+    envelopeFind,
+    polygonFind
+};
+
+
```
Comment (jedjas12) [inline db/api/v1/controllers/geometrySearch.controller.js]: For shame. Remove new line

File: db/api/v1/controllers/__tests__/geometrySearch.controller.test.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    });
+
+    it('catches database access exceptions to pass them to the error handler', async () => {
+        const DB_ERROR = new Error();
+        geomDb.polygonFind = jest.fn(async () => { throw DB_ERROR; });
+        await polygonFind(req, res, next);
+        expect(next).toHaveBeenCalledWith(DB_ERROR);
+        expect(res.json).not.toHaveBeenCalled();
+    });
+});
```
Comment (jedjas12) [inline db/api/v1/controllers/__tests__/geometrySearch.controller.test.js]: \n

File: db/api/v1/db/assets.db.js [hunk 4]
```diff
@@ -88,7 +89,7 @@ const createAsset = async (client, projectId, assetTypeId, location) => {
     `;
 
     // Generate the values to subsitute into the SQL command
-    const values = [projectId, assetTypeId, location.lattitude, location.longitude];
+    const values = [projectId, assetTypeId, location.longitude, location.latitude];
```
Comment (jedjas12) [inline db/api/v1/db/assets.db.js]: Make sure latitude (lattitude) still works

File: db/api/v1/routes/validate.js [hunk 5]
```diff
@@ -118,6 +118,64 @@ const extractBodyParam = (req, paramName) => req.body[paramName];
...
+/**
+ * Parses a list of latitude and longitude points.
+ * @param {Array} coordinateList - a list of latitude and longitude points.
+ * @returns {ParseResult} parse success if a valid list of latitude and longitude points of at least length 3
+ */
+const parseCoordinates = (coordinateList) => {
+   
+    // There needs to be at least 3 
+    if (coordinateList.length < 3)
+        return ParseResult.failure('Expected at least 3 points in the coordinates list');
```
Comment (jedjas12) [inline db/api/v1/routes/validate.js]: Create code block (instead of one liner)

File: db/api/v1/routes/validate.js [hunk 6]
```diff
@@ -160,6 +248,19 @@ const parseProjectName = (name) => {
...
+/**
+ * Validates if a radius string is a valid intenger.
+ * 
+ * @param {string} radiusString - the string to validate
+ * @returns {number} - the radius as an integer.
+ */
+const parseRadius = (radiusString) => {
+    const radius = parseInt(radiusString, 10);
+    return (isNaN(radius) || !utils.db.isValidDbInteger(radius))
+        ? ParseResult.failure(`Expected a number between 1 and ${utils.db.DB_INTEGER_MAX}`)
```
Comment (jedjas12) [inline db/api/v1/routes/validate.js]: Move ? for consistent styling (Anthony)

File: db/api/v1/utils/db.utils.js [hunk 7]
```diff
@@ -47,5 +76,6 @@ module.exports = {
     isValidDbInteger,
     beginTransaction,
     commitTransaction,
-    rollbackTransaction
+    rollbackTransaction,
+    makeLineString
 };
```
Comment (jedjas12) [inline db/api/v1/utils/db.utils.js]: \n

**Pull request merged**